src/OpenSage.FileFormats/EnumUtility.cs

```diff
@@ -26,8 +27,8 @@ public static bool IsValueDefined<TEnum>(TEnum value)
     }
 
     public static int GetEnumCount<TEnum>()
-        where TEnum : Enum
+        where TEnum : struct, Enum
     {
-        return Enum.GetNames(typeof(TEnum)).Length;
+        return Enum.GetValuesAsUnderlyingType<TEnum>().Cast<int>().Max() + 1;
```

Comment:
What is this method supposed to do? When I read `GetEnumCount` I assume 'get the number of items in the enum", which the old implementation did and this doesn't. How is this intended to work for flags enums?

Reply:
It gets the size of the array that is needed if you want to be able to use the enum values as integer indices into the array... which is subtely different from the number of items in the enum, if for example some values are repeated, or there's a non-contiguous range of values. I'll rename the method.

Reply:
> How is this intended to work for flags enums?

It doesn't work for flags enums.

Reply:
Ah, that makes a lot more sense for the behavior - yeah I think a rename would be good.